engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -690,6 +702,38 @@ yield switch (op.codeRepr()) {
       case Tree.Group group -> {
         yield translateExpression(group.getBody(), moreArgs, insideTypeSignature, isMethod);
       }
+      case Tree.TextLiteral txt -> {
+        var fullTxt = txt.codeRepr().trim();
+        var t = fullTxt.substring(1, fullTxt.length() - 1);
+        yield new IR$Literal$Text(t, getIdentifiedLocation(txt), meta(), diag());
+      }
+      case Tree.Arrow arrow when insideTypeSignature -> {
+        List<IR.Expression> args = nil();
+        Tree treeBody = null;
+        {
```

Comment:
that looks spurious?

Reply:
It hides the `head` variable from the rest of the scope.